dask/array/core.py

```diff
@@ -1247,8 +1252,22 @@ def chunksize(self):
     def dtype(self):
         return self._meta.dtype
 
+    @property
+    def _chunks(self):
+        return self.__chunks
+
+    @_chunks.setter
+    def _chunks(self, chunks):
+        self.__chunks = chunks
+
+        # When the chunks changes the cached properties that was dependent
+        # on it needs to be deleted:
+        for v in ["shape"]:
+            if v in self.__dict__:
+                del self.__dict__[v]
+
     def _get_chunks(self):
-        return self._chunks
+        return self.__chunks
```

Comment:
Does this have to be `self.__chunks` or can it be `self._chunks`?

Reply:
Sure I tried both and had trouble deciding which one to use. I just took the one with the shortest path to the value.

Reply:
I mildly prefer `self._chunks`.

Reply:
Done.